### PR TITLE
STYLE: Prepare for clang format style enforcement

### PR DESCRIPTION
**Pull request by hjmjohnson**
Provide initial code changes to facilitate enforcement of an ITK style that can be uniformly applied in an automated way.

These problems appeared when enforcing a consistent style with the clang-formatting tool.  It is a first step towards providing an easier code formatting style enforcement throughout ITK.

### CODE REVIEW

**Comment by blowekamp** [comment]
LGTM.

File: Testing/Temporary/CTestCostData.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+---
```
Comment (jhlegarreta) [inline Testing/Temporary/CTestCostData.txt]: Looks like this file accidentally slipped into the commit.

It has also [happened in the past](https://github.com/InsightSoftwareConsortium/ITK/commit/085907f8cc487e28925e2c1469ccbd4196242a9a#diff-61b29af89ba6a8e5ca026504561932db).

File: Testing/Temporary/LastTest.log [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Start testing: Jul 14 10:51 CDT
+----------------------------------------------------------
+End testing: Jul 14 10:51 CDT
```
Comment (jhlegarreta) [inline Testing/Temporary/LastTest.log]: Same as with `Testing/Temporary/CTestCostData.txt`.

I'd dare to say that the whole `Temporary` folder should not be committed.

File: Modules/IO/RAW/include/itkRawImageIO.hxx [hunk 3]
```diff
@@ -206,50 +265,54 @@ void RawImageIO< TPixel, VImageDimension >
...
-    else if itkWriteRawBytesAfterSwappingMacro(int, INT)
-    else if itkWriteRawBytesAfterSwappingMacro(long, LONG)
-    else if itkWriteRawBytesAfterSwappingMacro(unsigned long, ULONG)
-    else if itkWriteRawBytesAfterSwappingMacro(float, FLOAT)
-    else if itkWriteRawBytesAfterSwappingMacro(double, DOUBLE)
+    if ( this->GetComponentType() == USHORT )
+    {
+      itkWriteRawBytesAfterSwappingMacro< unsigned short >(
+        buffer, file, m_ByteOrder, numberOfBytes, numberOfComponents );
+    }
```
Comment (jhlegarreta) [inline Modules/IO/RAW/include/itkRawImageIO.hxx]: Is this the bracket indentation style that the rules will enforce? If yes, I'm OK. Previous conventions recommended the style that the removed lines are showing.

Comment (hjmjohnson) [reply Modules/IO/RAW/include/itkRawImageIO.hxx]: This will be the new style.

File: Testing/Temporary/CTestCostData.txt (hunk 1 above)
Comment (hjmjohnson) [reply Testing/Temporary/CTestCostData.txt]: Removed and .gitignore added.

File: Testing/Temporary/LastTest.log (hunk 2 above)
Comment (hjmjohnson) [reply Testing/Temporary/LastTest.log]: Removed and .gitignore added.

File: Modules/IO/RAW/include/itkRawImageIO.hxx [hunk 4]
```diff
@@ -139,35 +184,49 @@ void RawImageIO< TPixel, VImageDimension >
...
+  }
+  else if ( this->GetComponentType() == INT )
+  {
+    itkReadRawBytesAfterSwappingMacro< int >( buffer, m_ByteOrder, numberOfComponents );
+  }
+  else if ( this->GetComponentType() == LONG )
+  {
+    itkReadRawBytesAfterSwappingMacro< long >( buffer, m_ByteOrder, numberOfComponents );
+  }
+  else if ( this->GetComponentType() == ULONG )
```
Comment (dzenanz) [inline Modules/IO/RAW/include/itkRawImageIO.hxx]: ULONGLONG and LONGLONG are missing here. Perhaps add them in another commit?

File: Modules/IO/RAW/include/itkRawImageIO.hxx [hunk 5]
```diff
@@ -206,50 +265,54 @@ void RawImageIO< TPixel, VImageDimension >
...
+    else if ( this->GetComponentType() == LONG )
+    {
+      itkWriteRawBytesAfterSwappingMacro< long >( buffer, file, m_ByteOrder, numberOfBytes, numberOfComponents );
+    }
+    else if ( this->GetComponentType() == ULONG )
+    {
+      itkWriteRawBytesAfterSwappingMacro< unsigned long >(
+        buffer, file, m_ByteOrder, numberOfBytes, numberOfComponents );
+    }
+    else if ( this->GetComponentType() == FLOAT )
```
Comment (dzenanz) [inline Modules/IO/RAW/include/itkRawImageIO.hxx]: Again LONGLONG and ULONGLONG are missing. We probably forgot to add them when here we originally introduced them.

File: Modules/IO/RAW/include/itkRawImageIO.hxx [hunk 6]
```diff
@@ -101,6 +101,51 @@ void RawImageIO< TPixel, VImageDimension >
   m_ManualHeaderSize = true;
 }
 
+template < typename TStrongType >
+static void
+itkWriteRawBytesAfterSwappingMacro( const void * buffer, std::ofstream & file, ImageIOBase::ByteOrder byteOrder,
```
Comment (dzenanz) [inline Modules/IO/RAW/include/itkRawImageIO.hxx]: This causes a warning `unused function template 'itkWriteRawBytesAfterSwappingMacro'`:
https://open.cdash.org/viewBuildError.php?type=1&buildid=6008907

**Pull request merged**